### PR TITLE
Angular account service completion

### PR DESCRIPTION
**Pull request by jacobdavis1**
# housingxyz :: pull-request

## change proposal

Drastically reworked what was in master's housing folder. This PR's site has a working login/logout flow with a landing page and splits the navbar into three modes depending on the roles. Still needs testing and improvement but have you seen what's up there now? Currently includes work from Shawn and Alberto and is expecting work from Gabe, Marielle, and Wesley.

Also includes a working account service and notification service in the Angular side. My hope is tomorrow we can get the whole flow working demonstrably before the code freeze.

Yes, it is large. Very large. It was sort of like eating jambalaya, but the rice was code and so was the soup and the vegetables.

## reviewer

- @escalonn

## contributing

this project is maintained under the terms of the [Contribution Guidelines][contribution-guidelines-url].

[contribution-guidelines-url]: https://github.com/housingxyz/housingxyz/blob/master/.github/CONTRIBUTING.md 'CONTRIBUTION GUIDELINES'

### CODE REVIEW

**Comment by jacobdavis1** [comment]
We can recover some of the commented out tests after the code gets in.

**Comment by escalonn** [comment]
for the record, just replacing the `it(...)` with `xit(...)` (x for exclude) makes the test skipped and not run, better way than commenting out

File: housing/src/app/app-routing.module.ts [hunk 1]
```diff
@@ -1,36 +1,20 @@
 import { NgModule } from '@angular/core';
```
Comment (gpyofavv) [inline housing/src/app/app-routing.module.ts]: Thank you for this, Greg's team is unaffected by these much-needed deletions.

File: housing/src/app/app.module.ts [hunk 2]
```diff
@@ -5,13 +5,9 @@ import { AppRoutingModule } from './app-routing.module';
 import { AppComponent } from './app.component';
```
Comment (gpyofavv) [inline housing/src/app/app.module.ts]: Greg's team is unaffected.

File: housing/src/app/coordinator-notifications/notification-details/notification-details.component.html [hunk 3]
```diff
@@ -1,67 +1,6 @@
 <div class="outer">
```
Comment (gpyofavv) [inline housing/src/app/coordinator-notifications/notification-details/notification-details.component.html]: Near-total deletion of your own html?😕

File: housing/src/app/coordinator-notifications/notification-details/notification-details.component.spec.ts [hunk 4]
```diff
@@ -2,6 +2,7 @@ import { async, ComponentFixture, TestBed } from '@angular/core/testing';
```
Comment (gpyofavv) [inline housing/src/app/coordinator-notifications/notification-details/notification-details.component.spec.ts]: Comment or xit?

File: housing/src/app/manage-complex/add-complex/add-complex.component.html [hunk 5]
```diff
@@ -186,24 +186,16 @@ <h3>Add a Living Complex</h3>
...
-        <p>
-          Phone Number: {{ currentProvider.apiTrainingCenter.contactNumber }}
-        </p>
-        <p>{{ currentProvider.apiTrainingCenter.apiAddress.streetAddress }}</p>
-        <p>
-          {{ currentProvider.apiTrainingCenter.apiAddress.city }},
-          {{ currentProvider.apiTrainingCenter.apiAddress.state }}
-          {{ currentProvider.apiTrainingCenter.apiAddress.zipcode }}
-        </p>
+        <!-- Hardcode it -->
```
Comment (gpyofavv) [inline housing/src/app/manage-complex/add-complex/add-complex.component.html]: Just to confirm, you want this hard coded?

File: housing/src/app/manage-complex/add-complex/add-complex.component.ts [hunk 6]
```diff
@@ -16,9 +17,12 @@ export class AddComplexComponent implements OnInit {
   // the values to the provider object are set on initialization
   currentProvider: Provider;
 
+  // TODO: POPULATE THIS
```
Comment (gpyofavv) [inline housing/src/app/manage-complex/add-complex/add-complex.component.ts]: To confirm, you want your TODO in master?

File: housing/src/app/manage-complex/add-room/add-room.component.spec.ts [hunk 7]
```diff
@@ -6,11 +6,11 @@ import { MatFormFieldModule } from '@angular/material/form-field';
 import { MatSelectModule } from '@angular/material/select';
 import { MatInputModule } from '@angular/material';
 import { NoopAnimationsModule } from '@angular/platform-browser/animations';
-import { TestServiceData } from 'src/app/services/static-test-data';
 import { AddRoomComponent } from './add-room.component';
 import { By } from '@angular/platform-browser';
+import { Complex } from 'src/interfaces/complex';
 
-describe('AddRoomComponent', () => {
+/* describe('AddRoomComponent', () => {
```
Comment (gpyofavv) [inline housing/src/app/manage-complex/add-room/add-room.component.spec.ts]: Comment or xit?

File: housing/src/app/manage-complex/edit-complex/edit-complex.component.spec.ts [hunk 8]
```diff
@@ -4,8 +4,8 @@ import { MatFormFieldModule, MatSelectModule, MatTableModule  } from '@angular/m
 import { EditComplexComponent } from './edit-complex.component';
 import { NoopAnimationsModule } from '@angular/platform-browser/animations';
 import { By } from '@angular/platform-browser';
-import { TestServiceData } from 'src/app/services/static-test-data';
 
+/*
```
Comment (gpyofavv) [inline housing/src/app/manage-complex/edit-complex/edit-complex.component.spec.ts]: comment or xit?

File: housing/src/app/manage-complex/edit-room/edit-room.component.spec.ts [hunk 9]
```diff
@@ -12,8 +12,7 @@ import { By } from '@angular/platform-browser';
 import { EditRoomComponent } from './edit-room.component';
 import { Complex } from 'src/interfaces/complex';
 
-import { TestServiceData } from 'src/app/services/static-test-data';
-
+/*
```
Comment (gpyofavv) [inline housing/src/app/manage-complex/edit-room/edit-room.component.spec.ts]: comment or xit?

File: housing/src/app/app-routing.module.ts (hunk 1 above)
Comment (gpyofavv) [inline housing/src/app/app-routing.module.ts]: greg unaffected

File: housing/src/app/app.module.ts (hunk 2 above)
Comment (gpyofavv) [inline housing/src/app/app.module.ts]: greg unaffected

File: account/src/Revature.Account.Api/Startup.cs [hunk 10]
```diff
@@ -66,6 +66,7 @@ public void ConfigureServices(IServiceCollection services)
       {
         options.Authority = $"http://{Auth0Helper.Domain}/";
         options.Audience = Auth0Helper.Audience;
+        options.RequireHttpsMetadata = !Configuration.GetSection("Auth0").GetValue<bool>("IsDevelopment");
```
Comment (escalonn) [inline account/src/Revature.Account.Api/Startup.cs]: this is the part where itd be nice if it defaulted to false in the absence of a value, but fine

File: housing/src/app/app.module.ts [hunk 11]
```diff
@@ -102,7 +96,7 @@ import { ShowRoomComponent } from './manage-complex/show-room/show-room.componen
       multi: true
     }
   ],
-  entryComponents: [AmenityDialogueComponent, RequestDialogComponent],
+  entryComponents: [AmenityDialogueComponent],
```
Comment (escalonn) [inline housing/src/app/app.module.ts]: not sure that we need to declare any entrycomponents

File: housing/src/app/manage-complex/add-complex/add-complex.component.ts [hunk 12]
```diff
@@ -40,11 +44,13 @@ export class AddComplexComponent implements OnInit {
...
-        contactNumber: '',
-        companyName: '',
-        apiTrainingCenter: null
+        providerId: null,
+        coordinatorId: null,
+        name: '',
+        email: '',
+        status: null,
+        accountCreatedAt: new Date(),
+        accountExpiresAt: new Date()
```
Comment (escalonn) [inline housing/src/app/manage-complex/add-complex/add-complex.component.ts]: i don't understand why we are setting these values, neither here nor in the notification service

**Pull request merged**